ProjNet/CoordinateSystems/Transformations/MathTransform.cs

```diff
@@ -1,246 +1,521 @@
+        /// <param name="strideZ">A stride value for the z-ordinate series</param>
+        protected virtual void TransformCore(Span<double> xs, Span<double> ys, Span<double> zs,
+            int strideX, int strideY, int strideZ)
+        {
+            for (int i = 0, j = 0, k = 0; i < xs.Length; i += strideX, j += strideY, k += strideZ)
+            {
+                Transform(ref xs[i], ref ys[j], ref zs[k]);
+            }
+        }
+
+        private readonly double[] _dummyZ = new double[1];
+
+        /// <summary>
+        /// Core method to transform a series of points defined by their ordinates.
+        /// The transformation is performed in-place.
+        /// </summary>
+        /// <param name="xs">A series of x-ordinate values</param>
+        /// <param name="ys">A series of y-ordinate values</param>
+        /// <param name="strideX">A stride value for the x-ordinate series</param>
+        /// <param name="strideY">A stride value for the y-ordinate series</param>
+        /// <exception cref="ArgumentException">If the provided span and stride values don't result in matching number of ordinates.</exception>
+        public void Transform(Span<double> xs, Span<double> ys, int strideX = 1, int strideY = 1)
+        {
+            int elementsX = xs.Length / strideX + xs.Length % strideX != 0 ? 1 : 0;
+            int elementsY = ys.Length / strideY + ys.Length % strideY != 0 ? 1 : 0;
+
+            if (elementsX != elementsY)
+                throw new ArgumentException("Spans of ordinate values don't match in size.");
+
+            TransformCore(xs, ys, _dummyZ, strideX, strideY, 0);
+        }
+
+        /// <summary>
+        /// Core method to transform a series of points defined by their ordinates.
+        /// The transformation is performed in-place.
+        /// </summary>
+        /// <param name="xs">A series of x-ordinate values</param>
+        /// <param name="ys">A series of y-ordinate values</param>
+        /// <param name="zs">A series of z-ordinate values</param>
+        /// <param name="strideX">A stride value for the x-ordinate series</param>
+        /// <param name="strideY">A stride value for the y-ordinate series</param>
+        /// <param name="strideZ">A stride value for the z-ordinate series</param>
+        /// <exception cref="ArgumentException">If the provided span and stride values don't result in matching number of ordinates.</exception>
+        public void Transform(Span<double> xs, Span<double> ys, Span<double> zs,
+            int strideX = 1, int strideY = 1, int strideZ = 1)
+        {
+            int elementsX = xs.Length / strideX + xs.Length % strideX != 0 ? 1 : 0;
+            int elementsY = ys.Length / strideY + ys.Length % strideY != 0 ? 1 : 0;
+            if (elementsX != elementsY)
+                throw new ArgumentException("Spans of ordinate values don't match in size.");
+
+            if (zs == default)
```

Comment:
I would rather check `zs.IsEmpty` here since I'm pretty sure you can have zero-length spans pointing to non-null data.